public/app/features/explore/LogsVolumePanelList.test.tsx

```diff
@@ -0,0 +1,55 @@
+import { render, screen } from '@testing-library/react';
+import React from 'react';
+
+import { DataQueryResponse, LoadingState, EventBusSrv } from '@grafana/data';
+
+import { LogsVolumePanelList } from './LogsVolumePanelList';
+
+jest.mock('./Graph/ExploreGraph', () => {
+  const ExploreGraph = () => <span>ExploreGraph</span>;
+  return {
+    ExploreGraph,
+  };
+});
+
+function renderPanel(logsVolumeData?: DataQueryResponse) {
+  render(
+    <LogsVolumePanelList
+      absoluteRange={{ from: 0, to: 1 }}
+      timeZone="timeZone"
+      splitOpen={() => {}}
+      width={100}
+      onUpdateTimeRange={() => {}}
+      logsVolumeData={logsVolumeData}
+      onLoadLogsVolume={() => {}}
+      onHiddenSeriesChanged={() => null}
+      eventBus={new EventBusSrv()}
+    />
+  );
+}
+
+describe('LogsVolumePanelList', () => {
+  it('shows loading message', () => {
+    renderPanel({ state: LoadingState.Loading, error: undefined, data: [] });
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
+  });
+
+  it('shows short warning message', () => {
+    renderPanel({ state: LoadingState.Error, error: { data: { message: 'Test error message' } }, data: [] });
+    expect(screen.getByText('Failed to load log volume for this query')).toBeInTheDocument();
+    expect(screen.getByText('Test error message')).toBeInTheDocument();
+  });
+
+  it('shows long warning message', () => {
+    // we make a long message
+    const messagePart = 'One two three four five six seven eight nine ten.';
+    const message = messagePart + ' ' + messagePart + ' ' + messagePart;
+
+    renderPanel({ state: LoadingState.Error, error: { data: { message } }, data: [] });
+    expect(screen.getByText('Failed to load log volume for this query')).toBeInTheDocument();
+    expect(screen.queryByText(message)).not.toBeInTheDocument();
+    const button = screen.getByText('Show details');
+    button.click();
```

Comment:
with `fireEvent` from `'@testing-library/react'` :
```suggestion
    fireEvent.click(screen.getByRole('button', { name: 'Show details' }))
```

Reply:
Updated in 836895fb6f9939ef69fe532e948b2eb6bf8c0be8